packages/aws-cdk/test/api/cloudformation-deployments.test.ts

```diff
@@ -142,3 +162,718 @@ test('if toolkit stack cannot be found but SSM parameter name is present deploym
 
   expect(requestedParameterName!).toEqual('/some/parameter');
 });
+
+test('readCurrentTemplateWithNestedStacks() can handle non-Resources in the template', async () => {
+  const cfnStack = new FakeCloudformationStack({
+    stackName: 'OutputParamRoot',
+    stackId: 'StackId',
+  });
+  CloudFormationStack.lookup = (async (_, stackName: string) => {
+    switch (stackName) {
+      case 'OutputParamRoot':
+        cfnStack.template = async () => ({
+          Resources: {
+            NestedStack: {
+              Type: 'AWS::CloudFormation::Stack',
+              Properties: {
+                TemplateURL: 'https://www.magic-url.com',
+              },
+              Metadata: {
+                'aws:asset:path': 'one-output-one-param-stack.nested.template.json',
+              },
+            },
+          },
+        });
+        break;
+
+      case 'NestedStack':
+        cfnStack.template = async () => ({
+          Resources: {
+            NestedResource: {
+              Type: 'AWS::Something',
+              Properties: {
+                Property: 'old-value',
+              },
+            },
+          },
+          Parameters: {
+            NestedParam: {
+              Type: 'String',
+            },
+          },
+          Outputs: {
+            NestedOutput: {
+              Value: {
+                Ref: 'NestedResource',
+              },
+            },
+          },
+        });
+        break;
+
+      default:
+        throw new Error('unknown stack name ' + stackName + ' found');
+    }
+
+    return cfnStack;
+  });
+
+  const rootStack = testStack({
+    stackName: 'OutputParamRoot',
+    template: {
+      Resources: {
+        NestedStack: {
+          Type: 'AWS::CloudFormation::Stack',
+          Properties: {
+            TemplateURL: 'https://www.magic-url.com',
+          },
+          Metadata: {
+            'aws:asset:path': 'one-output-one-param-stack.nested.template.json',
+          },
+        },
+      },
+    },
+  });
+
+  pushStackResourceSummaries('OutputParamRoot',
+    stackSummaryOf('NestedStack', 'AWS::CloudFormation::Stack',
+      'arn:aws:cloudformation:bermuda-triangle-1337:123456789012:stack/NestedStack/abcd',
+    ),
+  );
+  pushStackResourceSummaries('NestedStack',
+    stackSummaryOf('NestedResource', 'AWS::Something',
+      'arn:aws:something:bermuda-triangle-1337:123456789012:property',
+    ),
+  );
```

Comment:
Why do we need this? When do we get the physical ID of a non-Nested Stack resource?